client/packages/common/src/ui/components/inputs/TextInput/NumericTextInput.tsx

```diff
@@ -43,14 +45,34 @@ export const NumericTextInput: FC<NumericTextInputProps> = React.forwardRef(
     const {
       options: { separator, decimal },
     } = useCurrency();
-    const [textValue, setTextValue] = useState(format(value ?? defaultValue));
+    const [textValue, setTextValue] = useState(
+      format(value ?? defaultValue, { minimumFractionDigits: decimalMin })
+    );
+    const isFirstRender = useRef(true);
+
+    const isInputIncomplete = useCallback(
+      (value: string) =>
+        new RegExp(
+          // Checks for a trailing `.` or a `0` (not necessarily immediately)
+          // after a `.`
+          `^\\d*\\.$|\\d*${RegexUtils.escapeChars(decimal)}\\d*0$`
+        ).test(value),
+      [decimal]
+    );
 
     useEffect(() => {
-      setTextValue(format(value));
-      // Excluding `format` from deps array, despite warning, as its not
-      // necessary (static method) and causes problems resulting in the text
-      // value not being updated correctly
-    }, [value]);
+      if (isFirstRender.current) {
+        // On first render, ensure number value is set from defaultValue prop
+        if (textValue && value === undefined) onChange(parse(textValue));
+        isFirstRender.current = false;
```

Comment:
This sends the default value out in onChange(), right? Why is the default value actually needed? can't the user manage the default value from the outside?

Reply:
The `defaultValue` is just something that's always been part of the number inputs. Yes, you could instantiate the `value` state with a defaultValue outside the component which would do the same thing. I just didn't want to break possible implementations which are using it this way.